Change search engine's URL

Tjololo12 opened the pull request:
ohnorobot.com's search page uses index.php not index.pl.

Tjololo12 committed:
Change search engine's URL

ohnorobot.com's search page uses index.php not index.pl.

linuxdaemon commented on plugins/xkcd.py:
```diff
@@ -22,7 +22,7 @@ def xkcd_info(xkcd_id, url=False):
 def xkcd_search(term):
     search_term = requests.utils.quote(term)
-    request = requests.get("http://www.ohnorobot.com/index.pl?s={}&Search=Search&"
+    request = requests.get("http://www.ohnorobot.com/index.php?s={}&Search=Search&"
```
It may be better to use `http://www.ohnorobot.com/` instead of `http://www.ohnorobot.com/index.php` so it uses whatever page the site uses as the default.

Tjololo12 replied:
Good call, updated

Tjololo12 committed:
Merge branch 'gonzobot' into gonzobot

Tjololo12 committed:
Removed explicit index page callout

Updated URL for search engine to remove specific index page reference to use whatever default is configured

linuxdaemon commented on plugins/xkcd.py:
```diff
@@ -13,7 +13,6 @@
 def xkcd_info(xkcd_id, url=False):
     """ takes an XKCD entry ID and returns a formatted string """
     request = requests.get("http://www.xkcd.com/" + xkcd_id + "/info.0.json")
-    request.raise_for_status()
```
Why do you remove the error handling?

Tjololo12 replied:
I haven't touched this since October, when I made the minor URL change. I'm not sure what happened here, but I haven't committed anything in here in a while.

linuxdaemon commented on plugins/xkcd.py:
```diff
@@ -45,5 +43,5 @@ def xkcd_url(match):
 @hook.command()
 def xkcd(text):
-    """<search term> - Search for xkcd comic matching <search term>"""
+    """xkcd <search term> - Search for xkcd comic matching <search term>"""
```
Also your docstring should not contain the command itself, that is added dynamically

Tjololo12 replied:
I haven't touched this since October, when I made the minor URL change. I'm not sure what happened here, but I haven't committed anything in here in a while.

Tjololo12 committed:
Made changes recommended by linuxdaemon

Tjololo12 committed:
Added functionality to disable tells entirely by nick

Tjololo12 committed:
Merge branch 'gonzobot' into gonzobot

Tjololo12 commented:
Gah I didn't switch branches before the other commit. The tell.py changes were supposed to be in a different branch.

linuxdaemon commented:
Please move the `tell.py` changes to a different PR.

linuxdaemon commented:
Please fix this PR, reopen it when you are ready
